addons/binding/org.openhab.binding.helios/src/main/java/org/openhab/binding/helios/handler/HeliosHandler213.java

```diff
@@ -0,0 +1,962 @@
+                                break;
+                            }
+                            case AUDIOLOOPTEST: {
+                                if (event.params.get("result").getAsString().equals("passed")) {
+                                    updateState(new ChannelUID(getThing().getUID(), AUDIO_LOOP_TEST), OnOffType.ON);
+                                } else if (event.params.get("result").getAsString().equals("failed")) {
+                                    updateState(new ChannelUID(getThing().getUID(), AUDIO_LOOP_TEST), OnOffType.OFF);
+                                } else {
+                                    updateState(new ChannelUID(getThing().getUID(), AUDIO_LOOP_TEST), UnDefType.UNDEF);
+                                }
+
+                                DateTimeType stampType = new DateTimeType(DATE_FORMATTER.format(date));
+                                updateState(new ChannelUID(getThing().getUID(), AUDIO_LOOP_TEST_STAMP), stampType);
+                                break;
+                            }
+                            case MOTIONDETECTED: {
+                                if (event.params.get("state").getAsString().equals("in")) {
+                                    updateState(new ChannelUID(getThing().getUID(), MOTION), OnOffType.ON);
+                                } else if (event.params.get("state").getAsString().equals("out")) {
+                                    updateState(new ChannelUID(getThing().getUID(), MOTION), OnOffType.OFF);
+                                } else {
+                                    updateState(new ChannelUID(getThing().getUID(), MOTION), UnDefType.UNDEF);
+                                }
+
+                                DateTimeType stampType = new DateTimeType(DATE_FORMATTER.format(date));
+                                updateState(new ChannelUID(getThing().getUID(), MOTION_STAMP), stampType);
+                                break;
+                            }
+                            case KEYPRESSED: {
+                                StringType valueType = new StringType(event.params.get("key").getAsString());
+                                updateState(new ChannelUID(getThing().getUID(), KEY_PRESSED), valueType);
+
+                                DateTimeType stampType = new DateTimeType(DATE_FORMATTER.format(date));
+                                updateState(new ChannelUID(getThing().getUID(), KEY_PRESSED_STAMP), stampType);
+                                break;
+                            }
+                            case KEYRELEASED: {
+                                StringType valueType = new StringType(event.params.get("key").getAsString());
+                                updateState(new ChannelUID(getThing().getUID(), KEY_RELEASED), valueType);
+
+                                DateTimeType stampType = new DateTimeType(DATE_FORMATTER.format(date));
```

Comment:
A lot of the code here is repeated. You could move it outside `switch` and reuse it in `case`s you need.